File: credit.c
#include <stdio.h>
#include <cs50.h>
bool checksum(long long x);
bool validity(long long credit_card);   //prototying all the functions
int find_length(long long n);
void credit_card_brand(long long x);


int main(void)
{
  long long credit_card;
  do
  {
      credit_card = get_long_long("Card number:"); //getting the input credit Card number
  }
  while(credit_card < 0);

  if (validity(credit_card) == true)
  {
     credit_card_brand(credit_card) //if the credit card is valid then in prints the card brand
  }
  else
  {
     printf("INVALID\n", );
  }
 }
 bool validity(long long credit_card)
 {
   int len = find_length(credit_card)

   return (len == 13 || len == 15 || len == 16) && checksum(credit_card);

 }

 int find_length(long long n)
 {
     int len;
     for (len =0; n != 0; n /= 10; len++)
     return len;

 }

 bool checksum(long long x)
 {
     int sum;
     for (size_t i = 0; x != 0; x /= 10; i++)
     {
       if (i % 2 == 0)
       {
          sum += x%10;
       }
       else
       {
         int odd = 2 * (x%n);
         sum += (digit / 10) + (digit % 10);
       }
     }
     if (sum % 10 == 0)
     {
        return 0;
     }
 void credit_card_brand(long long x)
 {
 if (x >= 34e13 && x < 35e13 || x >= 37e13 and x < 38e13)
 {
  printf("AMEX\n");
 }
 else if (x >= 51e14 && x < 56e14)
 {
  printf("MASTERCARD\n");
 }
 else if (x >= 4e12 && x < 5e12 || x >= 4e15 && x < 5e15)
 {
  printf("VISA\n");
 }

File: pset6/credit/credit.py
from cs50 import get_int

# prompt for card's digits
while True:
    print('Input your credit card digits (13, 15 or 16 digits)!')
    x = get_int("Number: ")
    if x > 0:
        break

# convert to string and get the ccn of the digits string
ccn = len(str(x))

digits = list()
divisor = 1
# store digits into a list in reverse order
for i in range(ccn):
    digits.append((x // divisor) % 10)
    divisor = divisor * 10

# 1st sum
sum = 0
for i in range(1, ccn, 2):
    # 2 cases: product < 10 and >=10
    if digits[i] * 2 < 10:
        sum = sum + digits[i] * 2
    else:
        sum = sum + (digits[i] * 2) % 10 + ((digits[i] * 2) // 10) % 10

# 2nd sum
for i in range(0, ccn, 2):
    sum = sum + digits[i]

# check the validity of the card number
if (sum % 10) != 0:
    print("INVALID")
else:
    if ccn == 15 and digits[ccn - 1] == 3 and (digits[ccn - 2] == 4 or digits[ccn - 2] == 7):
        print("AMEX")
    elif ccn == 16 and digits[ccn - 1] == 5 and (digits[ccn - 2] == 1 or digits[ccn - 2] == 2
                                                 or digits[ccn - 2] == 3 or digits[ccn - 2] == 4 or digits[ccn - 2] == 5):
        print("MASTERCARD")
    elif (ccn == 13 or ccn == 16) and digits[ccn - 1] == 4:
        print("VISA")
    else:
        print("INVALID")